pantos/servicenode/blockchains/ethereum.py

```diff
@@ -340,6 +340,17 @@ def __create_node_connections(self) -> NodeConnections:
         provider_timeout = self._get_config()['provider_timeout']
         return self._get_utilities().create_node_connections(provider_timeout)
 
+    def __get_nonce(self, node_connections: NodeConnections,
+                    internal_transfer_id: int) -> int:
+        transaction_count = node_connections.eth.get_transaction_count(
+            self.__address).get_maximum_result()
+        database_access.update_transfer_nonce(internal_transfer_id,
+                                              self.get_blockchain(),
+                                              transaction_count)
+        nonce = database_access.read_transfer_nonce(internal_transfer_id)
```

Comment:
This pattern is not new here, I just extracted that code to a separate function for better readability. Nonetheless, I think there was a good reason to split this up when we implemented this a while ago. I cannot remember the exact reason right now, but it was somehow caused by the complex `UPDATE` statement we have here. If you want, we can investigate again in a separate ticket if we actually could add a `RETURNING` clause to the `UPDATE` statement (which would then also affect our other off-chain projects, where we use a similar pattern), although I don't think it's too important right now.

Reply:
IIRC the only way to run the update nonce query was using 'synchronize_session': False.

If you use this option, you cannot return the updated nonce at the end of the transaction (because the transfer object in the session is not synced with the actual fields from the db)